### PR TITLE
Prototype for chunk manipulation function

### PR DESCRIPTION
**Pull request by mkindahl**
Adding a function drop_chunk that can be used to delete an individual
chunk from a hypertable to allow us to experiment with per-chunk
manipulation functions.

The commit also add a simplistic view for experimenting with that
provide time constraints on the chunks and can be used to select
chunks.

### CODE REVIEW

File: sql/views.sql [hunk 1]
```diff
@@ -25,6 +25,23 @@ CREATE OR REPLACE VIEW timescaledb_information.hypertable_size_info AS
       CASE WHEN has_schema_privilege(ht.schema_name,'USAGE') THEN format('%I.%I',ht.schema_name,ht.table_name) ELSE NULL END
     ) bsize ON true;
 
+-- View to provide some information about chunks
+-- chunk_id     Chunk ID
+-- lower_bound  Lower time constraint bound for chunk
+-- upper_bound  Upper time constraint bound for chunk
+-- hypertable   Hypertable that the chunk belongs to
+CREATE OR REPLACE VIEW timescaledb_information.chunks AS
+SELECT format('%1$I.%2$I', ch.schema_name, ch.table_name)::regclass AS chunk_id
```
Comment (erimatnor) [inline sql/views.sql:34]: These two IDs have confusing names given that we have internal IDs too. Should probably be something like chunk_oid or cunk_relid.

File: sql/views.sql [hunk 2]
```diff
@@ -25,6 +25,23 @@ CREATE OR REPLACE VIEW timescaledb_information.hypertable_size_info AS
...
+     , format('%1$I.%2$I', ht.schema_name, ht.table_name)::regclass AS hypertable_id
+     , tstzrange(TIMESTAMPTZ 'epoch' + sl.range_start * INTERVAL '1 microsecond',
+                 TIMESTAMPTZ 'epoch' + sl.range_end * INTERVAL '1 microsecond') AS time_range
+  FROM _timescaledb_catalog.chunk ch
+  JOIN _timescaledb_catalog.hypertable ht ON ch.hypertable_id = ht.id
+  JOIN _timescaledb_catalog.dimension di ON di.hypertable_id = ht.id
+  JOIN _timescaledb_catalog.chunk_constraint cn ON cn.chunk_id = ch.id
+  JOIN _timescaledb_catalog.dimension_slice sl ON cn.dimension_slice_id = sl.id
+ WHERE column_type = 'timestamptz'::regtype;
```
Comment (erimatnor) [inline sql/views.sql:43]: Looks like this will only show chunks of hypertables that have a timestamptz time column. Not sure this is what we'd like to do.

File: src/chunk.c [hunk 3]
```diff
@@ -3240,6 +3245,15 @@ ts_chunk_drop_process_materialization(Oid hypertable_relid,
 /* Continuous agg materialization hypertables can be dropped
  * only if a user explicitly specifies the table name
  */
+Datum
+ts_pg_chunk_drop(PG_FUNCTION_ARGS)
+{
+	int32 chunk_id = PG_GETARG_INT32(0);
```
Comment (erimatnor) [inline src/chunk.c:3251]: Bad cast. This is a regclass, not int32.

File: src/chunk.c [hunk 4]
```diff
@@ -3240,6 +3245,15 @@ ts_chunk_drop_process_materialization(Oid hypertable_relid,
...
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_id, 0, true);
+	ts_chunk_drop(chunk, DROP_RESTRICT, DEBUG1);
+	PG_RETURN_INT32(chunk_id);
```
Comment (erimatnor) [inline src/chunk.c:3254]: First, this has the wrong return type. Second, I don't think it makes sense to return the OID (which I belive is your intention) of an object that no longer exists. Should probably return the fully qualified name, or a boolean, if anything.

File: sql/CMakeLists.txt [hunk 5]
```diff
@@ -21,35 +21,35 @@ set(IMMUTABLE_API_SOURCE_FILES
 
 # The rest of the source files defining mostly functions
 set(SOURCE_FILES
-  pre_install/types.functions.sql
-  hypertable.sql
+  bgw_scheduler.sql
```
Comment (erimatnor) [inline sql/CMakeLists.txt:24]: These changes seem unrelated to PR.

File: src/chunk.c [hunk 6]
```diff
@@ -2284,14 +2285,18 @@ ts_chunk_get_by_relid(Oid relid, bool fail_if_not_found)
 }
 
 Chunk *
-ts_chunk_get_by_id(int32 id, bool fail_if_not_found)
+ts_chunk_get_by_id(int32 chunk_id, bool fail_if_not_found)
```
Comment (erimatnor) [inline src/chunk.c:2288]: More unrelated changes.

**Comment by k-rus** [comment]
@mkindahl Can you add tests with space dimensions?

**Comment by mkindahl** [comment]
Accidentally closed the wrong PR, reopening.

File: sql/views.sql (hunk 2 above)
Comment (mkindahl) [reply sql/views.sql:43]: It was only intended as an example for how a view could be used with the function, not as a final solution.

File: sql/CMakeLists.txt (hunk 5 above)
Comment (mkindahl) [reply sql/CMakeLists.txt:24]: I sorted the files to make is easier to find duplicates. Unsorted it again.

File: sql/views.sql (hunk 1 above)
Comment (mkindahl) [reply sql/views.sql:34]: I actually think it the name is fine (there is no risk for confusion and it happen to be a reference to a chunk), but it could be something like `chunk_id` since it is just an identifier and any connection to other tables is a coincidence. If we use `regclass` it is an OID, but if we decide for some other representation, we will have to change the name again. Not sure if we have any agreement on this.

Called it `_oid` in the meantime since that seem to be close to what we discussed last. We can change later if we want.

File: sql/views.sql (hunk 2 above)
Comment (mkindahl) [reply sql/views.sql:43]: I replaced this with a view that is used just for testing. We can take the "real" view in a different discussion.

File: src/chunk.c (hunk 6 above)
Comment (mkindahl) [reply src/chunk.c:2288]: Reverted.

File: src/chunk.c (hunk 3 above)
Comment (mkindahl) [reply src/chunk.c:3251]: Fixed.

File: src/chunk.c (hunk 4 above)
Comment (mkindahl) [reply src/chunk.c:3254]: Returns a string containing the (fully qualified) chunk name. Void could work as well, but this seems to be the most useful value to return.

**Comment by mkindahl** [comment]
Experimental. Discarding.
